### DISCUSSION TITLE
Feature Request: try_decode(blob)

### DISCUSSION THREAD

**Discussion by orf** [duckdb/duckdb#15284] [Ideas]
There is currently no way I can see to attempt to convert a blob column to a varchar, and return NULL if it's not valid utf8.

There is the [decode blob function](https://duckdb.org/docs/sql/functions/blob.html#decodeblob), but this outright fails if the input is not utf8 rather than returning null.

This is useful if you have a column with text and binary content, and you only want to select the text content.

**Comment by chapmanjacobd** [comment]
You can use `TRY_CAST(column_name AS VARCHAR)`:

https://duckdb.org/docs/sql/data_types/typecasting.html#casting-operations-matrix

If it doesn't work, maybe it's a bug: https://github.com/duckdb/duckdb/issues/13097

**Comment by casperhart** [comment]
This is something I would like to see as well.
@chapmanjacobd casting and decoding have different behaviour, so `try_cast` is not a valid option, for example:

```
D select decode('\xC3\xBC'::BLOB) as decode, try_cast('\xC3\xBC' as varchar) as try_cast;
┌─────────┬──────────┐
│ decode  │ try_cast │
│ varchar │ varchar  │
├─────────┼──────────┤
│ ü       │ \xC3\xBC │
└─────────┴──────────┘
```

Perhaps a `try_decode()`, or even better if there were multiple options in `decode()` for how to handle errors like in python:
- `errors = "abort"` (default, fail entirely)
- `errors = "strict"` (return NULL)
- `errors = "ignore"` (skip invalid characters)
- `errors = "replace"` (replace invalid characters with a placeholder)

Currently I have a python UDF to handle this, but it would be much cleaner if I could do it in duckdb natively.

**Reply by soerenwolfers** [reply]
Your example is missing a cast to blob in the second example. As it stands, you're converting string to string, so of course your getting the original string back.

**Reply by casperhart** [reply]
@soerenwolfers is this what you mean?

```
D select decode('\xC3\xBC'::BLOB) as decode, try_cast('\xC3\xBC'::BLOB as varchar) as try_cast;
┌─────────┬──────────┐
│ decode  │ try_cast │
│ varchar │ varchar  │
├─────────┼──────────┤
│ ü       │ \xC3\xBC │
└─────────┴──────────┘
```
If not, could you please give an example? If it's possible to do this with existing functionality that would be great
